File: pset6/readability.py
from cs50 import get_string

def main():
    count = 0
    l_count = 0
    w_count = 1
    s_count = 0

    text = get_string("Text: ")

    for c in text:
        if c.isalpha():
            l_count += 1
        elif c == " ":
            w_count += 1
        elif c == "." or c == "!" or c == "?":
            s_count += 1

    grade = round(0.0588 * (100 * l_count / w_count) - 0.296 * (100 * s_count / w_count) - 15.8)

    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")

if __name__ == "__main__":
    main()

File: pset2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>
#include <ctype.h>

int main(void)
{
    string text = get_string("Text: ");
    int l_count = 0;
    int w_count = 1;
    int s_count = 0;

    // count no. of words
    for (int i = 0; i < strlen(text); i++)
    {
        // letters
        if ((tolower(text[i]) >= 'a' && tolower(text[i]) <= 'z'))
        {
            l_count++;
        }

        // words
        else if (text[i] == ' ')
        {
            w_count++;
        }

        // sentences
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            s_count++;
        }
    }

    // calculate grade
    float grade = 0.0588 * (100 * (float) l_count / (float) w_count) - 0.296 * (100 * (float) s_count / (float) w_count) - 15.8;

    // print grade
    if (grade >= 0 && grade < 16)
    {
        printf("Grade %i\n", (int) round(grade));
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Before Grade 1\n");
    }
}